sktime/transformations/series/subsequence_extraction.py

```diff
@@ -0,0 +1,194 @@
+"""Subsequence extraction transformer.
+
+A transformer for the extraction of contiguous subsequences of specified
+length based on maximal/minimal rolling window aggregates.
+"""
+
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+import warnings
+from functools import partial
+
+import numpy as np
+import pandas as pd
+from numpy._core._multiarray_umath import _ArrayFunctionDispatcher
+
+from sktime.transformations.base import BaseTransformer
+
+__all__ = ["SubsequenceExtractionTransformer"]
+__author__ = ["wirrywoo"]
+
+
+class SubsequenceExtractionTransformer(BaseTransformer):
+    r"""
+    Extract contiguous subsequences of specified length based on rolling aggregates.
+
+    A transformer for the extraction of contiguous subsequences of specified
+    length based on maximal/minimal rolling window aggregates.
+
+    Given a sequence :math:`\\{x_1, x_2, \cdots, x_n \\}` and `subseq_len` integer
+    :math:`k` such that :math:`0 < k \leq n`, the transformer's task is to find index
+    :math:`i` satisfying :math:`1 \leq i \leq i + k - 1 \leq n` such that for given
+    `aggregate_fn` :math:`A: \mathbb{R}^k \longrightarrow \mathbb{R}`:
+
+    1. :math:`A(x_{i}, \cdots, x_{i+k-1})` is maximal when `selector = 'max'`, and
+    2. :math:`A(x_{i}, \cdots, x_{i+k-1})` is minimal when `selector = 'min'`.
+
+    When `aggregate_fn = np.sum` and `selector = 'max'`, the problem degenerates to the
```

Comment:
to ensure the docstring displays code fonts correctly, use *double backticks*, not single backticks for code. Single backticks are reserved for rst expressions such as `:math: `.

Also, minor point about this sentence, I would lead with the special case: "Maximum sum subarray is a special case and can be obtained by setting ..." - if this is an important subcase, I would also mention it further up.